spotbugs/src/main/java/edu/umd/cs/findbugs/detect/MethodCallInOptionalOrElse.java

```diff
@@ -0,0 +1,149 @@
+
+    public static final String BUG_ID = "MOE_METHOD_INVOKED_IN_OPTIONAL_ORELSE";
+
+    private static final Logger LOG = LoggerFactory.getLogger(MethodCallInOptionalOrElse.class);
+
+    private static final String[] BOXING_TYPES = {
+        Byte.class.getName(),
+        Character.class.getName(),
+        Double.class.getName(),
+        Float.class.getName(),
+        Integer.class.getName(),
+        Long.class.getName(),
+        Short.class.getName(),
+    };
+
+    private static final String JAVA_UTIL_OPTIONAL_SLASHED_CLASS_NAME = "java/util/Optional";
+    private static final String JAVA_UTIL_OPTIONAL_SIGNATURE = "L" + JAVA_UTIL_OPTIONAL_SLASHED_CLASS_NAME + ";";
+    private static final String JAVA_UTIL_OPTIONAL_OR_ELSE_METHOD_NAME = "orElse";
+    private static final String JAVA_LANG_XXX_VALUE_OF_METHOD_NAME = "valueOf";
+
+    private final BugAccumulator bugAccumulator;
+
+    private XMethod calledMethod;
+    private boolean inJavaLangOptionalChain;
+
+    public MethodCallInOptionalOrElse(BugReporter bugReporter) {
+        bugAccumulator = new BugAccumulator(bugReporter);
+    }
+
+    @Override
+    public void visit(Code obj) {
+        super.visit(obj);
+        bugAccumulator.reportAccumulatedBugs();
+        reset();
+    }
+
+    @Override
+    public void sawOpcode(int seen) {
+        if (seen == Const.GETSTATIC) {
+            final XField field = getXFieldOperand();
+            LOG.debug("[{} {}] Load static field {} of type {}", inJavaLangOptionalChain, calledMethod,
+                    field.getName(), field.getSignature());
+            if (JAVA_UTIL_OPTIONAL_SIGNATURE.equals(field.getSignature())) {
+                enterJavaLangOptionalChain();
+            }
+        } else if (isRegisterLoad()) {
+            final LocalVariable local = getMethod().getLocalVariableTable().getLocalVariable(getRegisterOperand(), getNextPC());
+            LOG.debug("[{} {}] Load {} of type {}", inJavaLangOptionalChain, calledMethod,
+                    local.getName(), local.getSignature());
+            if (JAVA_UTIL_OPTIONAL_SIGNATURE.equals(local.getSignature())) {
+                enterJavaLangOptionalChain();
+            }
+        } else if (seen == Const.PUTSTATIC) {
+            final XField field = getXFieldOperand();
+            LOG.debug("[{} {}] Store static field {} of type {}", inJavaLangOptionalChain, calledMethod,
+                    field.getName(), field.getSignature());
+            if (inJavaLangOptionalChain && JAVA_UTIL_OPTIONAL_SIGNATURE.equals(field.getSignature())) {
+                reset();
+            }
+        } else if (isRegisterStore()) {
+            final LocalVariable local = getMethod().getLocalVariableTable().getLocalVariable(getRegisterOperand(), getNextPC());
+            LOG.debug("[{} {}] Store {} of type {}", inJavaLangOptionalChain, calledMethod,
+                    local.getName(), local.getSignature());
```

Comment:
Tests are failing in the workflow due to `NullPointerException` https://github.com/spotbugs/spotbugs/runs/4491142063?check_suite_focus=true#step:5:322

```
  Exception analyzing edu.umd.cs.findbugs.detect.StringConcatenation using detector edu.umd.cs.findbugs.detect.MethodCallInOptionalOrElse
    java.lang.NullPointerException: Cannot invoke "org.apache.bcel.classfile.LocalVariable.getName()" because "local" is null
      At edu.umd.cs.findbugs.detect.MethodCallInOptionalOrElse.sawOpcode(MethodCallInOptionalOrElse.java:102)
```

Reply:
They all comes from a code like
```java
        try (var closable = ...) {
            // do something
        } catch (Exception ignored) {
        }
```
There is a hidden variable generated by the compiler for exceptions passed to the 'addSuppressed' method.
it seems that in our case such variables can be ignored.

Reply:
done